designs/2022-docs-information-architecture-update/README.md

```diff
@@ -0,0 +1,352 @@
+  - Hidden in sidebar
+- [MOVE PAGE] Architecture
+  - Current page: [Architecture](https://eslint.org/docs/latest/developer-guide/architecture/)
+  - Note: in a previous conversation with Nicholas, he mentioned to me that this page is pretty out-of date.
+  - There should probably be a future unit of work to refactor this page.
+  - However, it’s not included in the information architecture updates.
+- [MOVE & REFACTOR PAGE] Set up Development Environment
+  - Consolidate the pages [Getting the Source Code](https://eslint.org/docs/latest/developer-guide/source-code) and [Set up a Development Environment](https://eslint.org/docs/latest/developer-guide/development-environment)
+  - Getting the source code is basically just a step in the process of setting up the development environment. Therefore, these two pages should be consolidated
+  - Also the Directory Structure section on the [Getting the Source Code](https://eslint.org/docs/latest/developer-guide/source-code#directory-structure) page should be moved toward the bottom of the new Set up a Development Environment page.
+- [NEW PAGE(S)] Any other technical pages for maintainers
+  - I don’t have any thoughts on what these should be right now. I would defer to the core maintainers on what they think other maintainers should know.
+  - I think that these additional technical pages would make sense to include in **Phase 5: “Maintain ESLint” documentation updates**of [the documentation update project](https://github.com/eslint/eslint/issues/16365).
+  - NOTE: Maybe should be added, but not part of the initial IA project.
+- [MOVE PAGE] Run the Tests
+  - Current page: [Unit Tests](https://eslint.org/docs/latest/developer-guide/unit-tests)
+
+### New Sidebar (summary)
+
+With the above changes, the left-hand side navigation will look like:
+
+- Use ESLint in Your Project (landing page)
+  - Getting Started
+  - Core Concepts
+  - Configure
+    - Configuration Files (New)
+    - Configuration Files
+    - Configure Language Options
+    - Configure Rules
+    - Configure Plugins
+    - Ignore Code
+  - Command Line Interface Reference
+  - Rules Reference
+  - Integrations
+  - Migrate to v8.x
+- Extend ESLint (landing page)
+  - Create Plugins
+  - Custom Rules
+  - Custom Formatters
+  - Custom Parsers
+  - Custom Processors
+  - Share Configurations
+  - Node.js API Reference
+- Contribute to ESLint (landing page)
+  - Report Bugs
+  - Propose a New Rule
+  - Propose a Rule Change
+  - Request a Change
+  - Architecture
+  - Set up a Development Environment
+  - Run the Tests
+  - Work on Issues
+  - Submit a Pull Request
+  - Report Security Vulnerabilities
+  - Governance
+- Maintain ESLint (landing page)
+  - Manage Issues
+  - Review Pull Requests
+  - Manage Releases
+
+### HTTP Redirects
+
+The proposed changes would include the following HTTP redirects from current content locations to the locations proposed in the above outline.
```

Comment:
Thanks for putting this together. Some thoughts on the URL structure:

* There's an inconsistency between `/users`, `/extend`, and `/maintain`. It seems like we should be using the same word forms throughout. So, `/users`, `/extenders`, `/maintainers` or `/use`, `/extend`, `/maintain`, although I think I'd prefer `/using`, `/extending` and `/maintaining`. What do you think?
  * Whatever we decide, we should make it consistent all the way down so `/using/configuring` or `/use/configure` rather than `/users/configure`.
* I'm not sure if it's intentional or not, but there is a mix of `/users` and `/user` in the table below.

Reply:
Given the preference for removing avoiding gerunds, I think I'd lean toward for `/:persona/:topic`, e.g. `/contributors/pull-requests`, but I'd also be fine with the verb forms as suggested above.

Reply:
i absolutely agree with the point about consistency. good call out. 

i don't have a strong preference for the URL slug pattern.

`/:persona/:topic` as @btmills proposes has a 'RESTful' eloquence to it. however, the `:persona` could get a bit awkward with the persona names being a little clunky, such as `extenders` or `integrators`. but i think this is OK given that the context is a URL slug, not the text copy. just wanted to call this out. 

and to do verbs instead of persona (i.e `use`, `extend`, etc.) seems fine as well. though to have `use` in the slug in particular seems unclear b/c in a sense, everyone is using ESLint. but agian, it's in a URL slug, not the copy of the text, so i don't think the nuance here is too important.

so i guess tl;dr both options are fine w/ minor tradeoffs. 

i think i have a slight preference for the `:persona` based structure, but would defer to the core maintainers if you feel otherwise.

Reply:
Yeah, I feel like `/:persona` may be a bit too meta -- this seems like a leaky abstraction to me.

I think something around actions makes the most sense, though I do wonder if maybe we decouple those actions from the personas. For instance, what if a user came in and they went through the following pages:

1. `/use/setup` or `/use/install`
2. `/use/configure`

Although `/use/core-concepts` seems a bit weird to me.

@eslint/eslint-tsc what do you think?